### PR TITLE
Client with mspconfigs

### PR DESCRIPTION
**Pull request by 3100**

### CODE REVIEW

File: x/ibc/xx-fabric/types/header.go [hunk 1]
```diff
@@ -68,3 +91,73 @@ func (ci ChaincodeInfo) GetSignBytes() []byte {
...
+	prevID := ""
+
+	for _, mh := range mhs.Headers {
+		if err := mh.ValidateBasic(); err != nil {
+			return err
+		}
+		if m[mh.MSPID] {
+			return errors.New("some MSPHeaders are duplicated")
+		}
+		if prevID >= mh.MSPID {
```
Comment (3100) [inline x/ibc/xx-fabric/types/header.go:156]: define a compare function and use it.

File: x/ibc/xx-fabric/update.go [hunk 2]
```diff
@@ -121,5 +132,72 @@ func update(clientState ClientState, header Header) (ClientState, *ConsensusStat
...
+		newInfos = append(newInfos, types.NewMSPInfo(mh.MSPID, mh.Config, mh.Policy))
+	}
+
+	clientState.LastMSPInfos.Infos = newInfos
+	return clientState
+}
+
+func updateMSPConfig(clientState ClientState, mh types.MSPHeader) ClientState {
+	// assume idx != -1
+	idx := clientState.LastMSPInfos.IndexOf(mh.MSPID)
```
Comment (bluele) [inline x/ibc/xx-fabric/update.go:186]: It may be better to occur panic explicitly.

File: x/ibc/xx-fabric/update_test.go [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+	type args struct {
+		clientState ClientState
+		mspHeaders  types.MSPHeaders
+	}
+	tests := []struct {
+		name string
+		args args
+		want ClientState
+	}{
+		// TODO: Add test cases.
```
Comment (bluele) [inline x/ibc/xx-fabric/update_test.go:20]: Do you need this comment?

**Comment by bluele** [comment]
NOTE: We merge this PR for now, but we consider introducing an update sequence for security concerns such as replay attacks in header updates.

**Pull request merged**